### PR TITLE
Updated bloom_filter.hpp…

### PR DESCRIPTION
**Pull request by pmconrad**
from https://github.com/ArashPartow/bloom/blob/c89c2912c445dd5c0afe3d819ca012a6c6639e4e/bloom_filter.hpp

Most notably, the upstream license has changed to MIT, which should simplify #13

### CODE REVIEW

File: include/fc/bloom_filter.hpp [hunk 1]
```diff
@@ -11,16 +9,20 @@
...
- * http://www.opensource.org/licenses/cpl1.0.php                     *
+ * guidelines and in accordance with the MIT License.                *
+ * http://www.opensource.org/licenses/MIT                            *
  *                                                                   *
  *********************************************************************
 */
 
+
+#ifndef INCLUDE_BLOOM_FILTER_HPP
+#define INCLUDE_BLOOM_FILTER_HPP
```
Comment (jmjatlanta) [inline include/fc/bloom_filter.hpp:20]: Why the removal of #pragma once? I do not believe any compiler within our target platforms have a problem with it. IMHO these are a bit ugly.

File: include/fc/bloom_filter.hpp [hunk 2]
```diff
@@ -617,12 +660,21 @@ inline bloom_filter operator ^ (const bloom_filter& a, const bloom_filter& b)
 
 } // namespace fc
 
+FC_REFLECT( fc::bloom_filter, (salt_)(bit_table_)(salt_count_)(table_size_)(projected_element_count_)
+                              (inserted_element_count_)(random_seed_)(desired_false_positive_probability_) )
+FC_REFLECT( fc::bloom_parameters::optimal_parameters_t, (number_of_hashes)(table_size) )
+FC_REFLECT( fc::bloom_parameters, (minimum_size)(maximum_size)(minimum_number_of_hashes)
+                                  (maximum_number_of_hashes)(projected_element_count)
+                                  (false_positive_probability)(random_seed)(optimal_parameters) )
+
```
Comment (jmjatlanta) [inline include/fc/bloom_filter.hpp]: Is there a need for reflection here? I believe the bloom filter is only used for subscribing to notifications, so curious as to why this was added.

**Review by jmjatlanta — COMMENTED** [review]
Spacing in some areas are not in line with our standards, but I believe this is forgivable for 3rd party code. There are a few other comments below, which are more curiosity than pointing out issues. Changes look good. It works well on Ubuntu 18.04 / Boost 1.65.1

File: include/fc/bloom_filter.hpp (hunk 1 above)
Comment (pmconrad) [reply include/fc/bloom_filter.hpp:20]: Because upstream does it like this.

File: include/fc/bloom_filter.hpp (hunk 2 above)
Comment (pmconrad) [reply include/fc/bloom_filter.hpp]: No idea why I added this, will remove.

**Review by jmjatlanta — APPROVED** [review]
Looks good. Thanks Peter! Tested on Ubuntu 18.04 / Boost 1.65.1.

**Pull request merged**